apps/opik-guardrails-backend/opik_guardrails/services/validators/pii/validator.py

```diff
@@ -60,7 +65,7 @@ def validate(
             )
 
         return schemas.ValidationResult(
-            validation_passed=len(results) == 0,
+            validation_passed=validation_passed,
```

Comment:
From the original code it seems that the author meant to have it this way:
```suggestion
            validation_passed=len(grouped_results) == 0,
```

Anyway, the proposed solution will also work.